### PR TITLE
Error Recovery

### PR DESCRIPTION
**Pull request by wycats**
This pull request implements error recovery in the VM.

More information to come.

### CODE REVIEW

File: packages/@glimmer/debug/lib/generated/op-list.ts [hunk 1]
```diff
@@ -20,6 +19,8 @@ export const DebugOpList = [
   null,
   null,
   null,
+  null,
```
Comment (NullVoxPopuli) [inline packages/@glimmer/debug/lib/generated/op-list.ts:22]: what is the purpose of these nulls?

File: packages/@glimmer/opcode-compiler/lib/syntax/statements.ts [hunk 2]
```diff
@@ -259,29 +258,31 @@ STATEMENTS.add(SexpOpcodes.InElement, (op, [, block, guid, destination, insertBe
...
-STATEMENTS.add(SexpOpcodes.HandleError, (op, [, _handler, block, _inverse]) => {
-  return op.labels(() => [
-    // If there's an error inside `block`, jump to the catch block
-
-    op(Op.PushTryFrame, op.target('CATCH')),
+STATEMENTS.add(SexpOpcodes.HandleError, (op, [, handler, block, _inverse]) => {
+  op.labels(() => {
+    expr(op, handler);
+    op(Op.PushTryFrame, op.target('CATCH'));
+    // ReplayableIf(
```
Comment (NullVoxPopuli) [inline packages/@glimmer/opcode-compiler/lib/syntax/statements.ts:265]: don't forget that this is commented out!

**Comment by NullVoxPopuli** [comment]
I extracted the lint dependency updates to here: https://github.com/glimmerjs/glimmer-vm/pull/1474

**Comment by NullVoxPopuli** [comment]
rebase should make the diff significantly smaller

**Comment by NullVoxPopuli** [comment]
I'm going to be helping rebase, but:
A note on conflicting changes:
- dom-change-list is no longer in this repo, it's moved to another repo
- all the package.json scripts needed to change, due improper-for-ember configuration
- lint configs had to change due in order to work with CI and turbo
- various bugfixes in interfaces, reference, and runtime that were needed for ember-source

File: .eslintrc.cjs
Comment (chancancode) [inline .eslintrc.cjs:1]: What's up with this diff?

Comment (chancancode) [reply .eslintrc.cjs:1]: – and, in general, now that this is in (from what I understand) a ready for merge/review state, can you back out all these unrelated config/workflow changes into a separate PR and land them first or separately later? Don't want to merge a big experimental new feature (which has a non-zero chance of needed to be reverted later) with a lot of unrelated changes.

**Review by chancancode — CHANGES_REQUESTED** [review]
I don't think I'll be able to successfully review this in the current state in the amount of time I had allocated for the task. (To be fair, it was still marked as draft, but I was recently told it's ready.)

Let me know when it is cleaned up with the unrelated changes backed out; especially for the formatting changes, I would strongly prefer to have them landed ahead of time and this rebased on top (which, I think @NullVoxPopuli had already started doing).

**Comment by NullVoxPopuli** [comment]
Closing due to conflicts and https://github.com/glimmerjs/glimmer-vm/pull/1501 supersedes

**Pull request closed without merging**